### PR TITLE
Lookup models using the datasource modelbuilder instead of cfg file

### PR DESCRIPTION
**Pull request by sjschubert**
Reached this approach through a discussion with the loopback developers.  You can read the exchange on the google group:
https://groups.google.com/forum/#!searchin/loopbackjs/seth$20schubert/loopbackjs/dKOI4UHqfcU/JOoldrT38WQJ

### CODE REVIEW

File: tasks/loopback_auto.js [hunk 1]
```diff
@@ -95,47 +88,32 @@ module.exports = function(grunt) {
...
-          grunt.log.writeln(model + ' ' + colors.green(options.method));
-          dataSource[options.method](model, callback);
-        }
-        else {
-          logEvent('ignored');
-          grunt.log.writeln(model + ' ' + colors.grey('ignored') + ' -> DataSource: ' + colors.blue(config[model].dataSource));
-          goToNext();
-        }
+    //collect models for data source.
+    var models = _.select(ds.modelBuilder.models, function (m) {
```
Comment (paulomcnally) [inline tasks/loopback_auto.js]: change **m** with **model**

File: tasks/loopback_auto.js [hunk 2]
```diff
@@ -95,47 +88,32 @@ module.exports = function(grunt) {
...
+      return (m.dataSource === ds);
```
Comment (paulomcnally) [inline tasks/loopback_auto.js]: Use **model.dataSource**

File: tasks/loopback_auto.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ var _ = require('underscore');
 var path = require('path');
 var colors = require('colors');
 
-module.exports = function(grunt) {
+module.exports = function (grunt) {
```
Comment (paulomcnally) [inline tasks/loopback_auto.js]: I use google Code Style **function() {** don't **function () {**

File: tasks/loopback_auto.js [hunk 4]
```diff
@@ -95,47 +88,32 @@ module.exports = function(grunt) {
...
-          logEvent('ignored');
-          grunt.log.writeln(model + ' ' + colors.grey('ignored') + ' -> DataSource: ' + colors.blue(config[model].dataSource));
-          goToNext();
-        }
+    //collect models for data source.
+    var models = _.select(ds.modelBuilder.models, function (m) {
+      return (m.dataSource === ds);
+    });
+
+    _.each(models, function (m, idx, list) {
```
Comment (paulomcnally) [inline tasks/loopback_auto.js]: Use:

```
 models.forEach(function(model) {
    ...
    code
    ...
})
```

**Comment by sjschubert** [comment]
I've addressed the issues raised in the latest commit - let me know if you find anything else

**Comment by paulomcnally** [comment]
I will test today. Thanks

**Pull request closed without merging**